pkg/lib/authn/otp/service.go

```diff
@@ -166,7 +166,12 @@ func (s *Service) VerifyMagicLinkCode(code string, consume bool) (*Code, error)
 		return nil, err
 	}
 
-	if !secretcode.MagicLinkOTPSecretCode.Compare(codeModel.UserInputtedCode, codeModel.Code) {
+	userInputtedCode := codeModel.UserInputtedCode
```

Comment:
@carmenlau VerifyMagicLinkCode is currently overloaded with different use cases. Per our discussion, we should have one method for each use case.

Use case 1: Set the user inputted case in the approve login link flow.
Use case 2: Verify the user inputted case in the original sign in flow.